jax/lax/lax_parallel.py

```diff
@@ -690,11 +692,20 @@ def _axis_index_soft_pmap_rule(vals, mapped, chunk_size, *, axis_name):
 # wants to bind an axis name has to additionally implement `process_axis_index`
 # and put its main trace on the axis env stack.
 def _axis_index_bind(*, axis_name):
-  frame = core.axis_frame(axis_name)
-  if frame.main_trace is not None:
-    trace = frame.main_trace.trace_type(frame.main_trace, core.cur_sublevel())
-    return trace.process_axis_index(frame)
-  return core.Primitive.bind(axis_index_p, axis_name=axis_name)
+  if not isinstance(axis_name, tuple):
+    axis_name = (axis_name,)
+  inner_size = 1
+  index = 0
+  for name in reversed(axis_name):
+    frame = core.axis_frame(name)
+    if frame.main_trace is not None:
+      trace = frame.main_trace.trace_type(frame.main_trace, core.cur_sublevel())
+      name_idx = trace.process_axis_index(frame)
+    else:
+      name_idx = core.Primitive.bind(axis_index_p, axis_name=name)
+    index += name_idx * inner_size
+    inner_size *= psum(1, name)
+  return index
```

Comment:
I think we can instead leave the bind rule as it was, and just update the axis_index translation rule to actually use axis_name (and look it up in its axis_env argument).

Reply:
This change is not a fix for the bug, but adds the "Additionally, this adds support for querying the index along multiple
axes, which will be useful for gmap." part.

Reply:
Oh, I see. Thanks for explaining. I think that makes sense.

Reply:
So IIUC the idea here is to handle multiple axes in the bind rule, rather than the alternative of handling it in the translation rule and in `process_axis_index`. Is that right? That seems reasonable, though the alternative seems reasonable too. I think the `psum` primitive and its rules work more like the latter (i.e. allowing axis_name to be a list/tuple, ultimately calling into [this helper](https://github.com/google/jax/blob/997e56d0b2cd45fede4cdec117bb6520b8d0cec0/jax/interpreters/xla.py#L462) to handle it), but we don't need to be consistent with that.